migrate/migrate.go

```diff
@@ -0,0 +1,13 @@
+package migrate
+
+type Migrator interface {
+	Migrate() error
+}
+
+type ConnectionOptions struct {
```

Comment:
this really doesn't need to live in a separate file, ideally it should live close to where it is used, where possible.